### PR TITLE
Migrate Enzyme to RTL - /domains

### PR DESCRIPTION
**Pull request by ndv99**
## Done

Migrated the following files to RTL:
- [src/app/domains/components/RecordFields/RecordFields.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_domains?expand=1#diff-8b7ae5ab642023cbddb236a308a18d2d6d1c67051408acb892f451f66b93dd0b)
- [src/app/domains/views/DomainDetails/DomainDetails.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_domains?expand=1#diff-f0519d8b625d73e297ce8621fb109db521176cd3c731a1c5b65e1dae4cc5d16d)
- [src/app/domains/views/DomainDetails/DomainDetailsHeader/AddRecordForm/AddRecordForm.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_domains?expand=1#diff-4636e6070bc20d3c5e3d1f6d0fa4022c53733ff0c294e7f141c7a297a6b6e40d)
- [src/app/domains/views/DomainDetails/DomainDetailsHeader/DeleteDomainForm/DeleteDomainForm.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_domains?expand=1#diff-4491fadc948cdcd8a37d7b135873d3d6758f228b659f910870e1d8eee7fef2e8)
- [src/app/domains/views/DomainDetails/DomainDetailsHeader/DomainDetailsHeader.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_domains?expand=1#diff-e6655f592c710a6cec996c5431f17d1e26f2637e6a69ba6cac1d92e0c9d94b57)
- [src/app/domains/views/DomainDetails/DomainSummary/DomainSummary.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_domains?expand=1#diff-8bdcc8dc1de751d6d409723d617a35588ea657b8dd1105f01bac50f31415478d)
- [src/app/domains/views/DomainDetails/ResourceRecords/DeleteRecordForm/DeleteRecordForm.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_domains?expand=1#diff-f2bcfed8d3180d224828a91d14ebc21695757eda923b9e516b2604a79b800ed0)
- [src/app/domains/views/DomainDetails/ResourceRecords/EditRecordForm/EditRecordForm.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_domains?expand=1#diff-b40eea6e43c6de9e3087b9c31deabb693ac83f702eb1699e312649366bc20fef)
- [src/app/domains/views/DomainDetails/ResourceRecords/ResourceRecords.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_domains?expand=1#diff-58dd29d8ed6364dcf7729fbff618afa17d3cdee2a588c2d5a708a4b8faf544e2)
- [src/app/domains/views/DomainsList/DomainListHeader/DomainListHeader.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_domains?expand=1#diff-a9ce62c7b2f6a98dc34d5fda35c7b71abb57fa5d442f0bc7fbcd79ff67e0a5d3)
- [src/app/domains/views/DomainsList/DomainListHeaderForm/DomainListHeaderForm.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_domains?expand=1#diff-ad5aea348e80b0128f83c4decda4d242ed0170d33254c0b2a475e59e0c48c7fb)
- [src/app/domains/views/DomainsList/DomainsList.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_domains?expand=1#diff-a30ca7a883d5677fef7a1469c17d8c697d17d61ad1978ef0c1ca252edbc43e96)
- [src/app/domains/views/DomainsList/DomainsTable/DomainsTable.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:rtlMigration_domains?expand=1#diff-a4b130db4f69e376efff76a891fe1acb375082dc5f4c0476a896db18108d2690)

## Fixes

Fixes https://github.com/canonical/app-tribe/issues/1062.

### CODE REVIEW

File: src/app/domains/views/DomainDetails/DomainDetails.test.tsx [hunk 1]
```diff
@@ -21,19 +16,11 @@ describe("DomainDetails", () => {
...
-        <MemoryRouter
-          initialEntries={[{ pathname: "/domain/1", key: "testKey" }]}
-        >
-          <CompatRouter>
-            <Route exact path="/domain/:id" render={() => <DomainDetails />} />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
-    );
+    renderWithBrowserRouter(<DomainDetails />, {
```
Comment (huwshimi) [inline src/app/domains/views/DomainDetails/DomainDetails.test.tsx:19]: Nice!

File: src/app/domains/views/DomainDetails/DomainDetailsHeader/AddRecordForm/AddRecordForm.test.tsx [hunk 2]
```diff
@@ -1,45 +1,41 @@
...
-import { act } from "react-dom/test-utils";
+import { screen, render } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
 import { Provider } from "react-redux";
 import { MemoryRouter } from "react-router-dom";
 import { CompatRouter } from "react-router-dom-v5-compat";
 import configureStore from "redux-mock-store";
 
-import AddRecordForm from "./AddRecordForm";
+import { Labels as RecordFieldsLabels } from "../../../../components/RecordFields/RecordFields";
```
Comment (huwshimi) [inline src/app/domains/views/DomainDetails/DomainDetailsHeader/AddRecordForm/AddRecordForm.test.tsx:8]: This should probably be an absolute import.

**Review by huwshimi — APPROVED** [review]

**Comment by huwshimi** [comment]
@ndv99 oh, just realised this was a draft, if it wasn't meant to be review feel free to ignore and I'll do another review when it's ready.

**Comment by ndv99** [comment]
> @ndv99 oh, just realised this was a draft, if it wasn't meant to be review feel free to ignore and I'll do another review when it's ready.

Oh nah that's all good! It was definitely ready for review, I pushed it and created the pr at the end of the day yesterday but forgot to actually set it to ready after the checks passed!

File: src/app/domains/views/DomainDetails/DomainDetailsHeader/AddRecordForm/AddRecordForm.test.tsx (hunk 2 above)
Comment (ndv99) [reply src/app/domains/views/DomainDetails/DomainDetailsHeader/AddRecordForm/AddRecordForm.test.tsx:8]: I'll get that changed quickly. Think it's a couple places where that happens

**Pull request merged**